.github/workflows/lint-secrets.yml

```
@@ -0,0 +1,40 @@
+---
+name: "Lint secrets"
+
+on:
+  push:
+    branches:
+      - 'main'
+      - 'releases/**'
+  pull_request: {}
+
+jobs:
+  lint-secrets:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v3
+        with:
+          fetch-depth: 0
+      # TODO(danh): this install+configure git-secrets is a good candidate for a
+      # custom action repo -- closest to our needs I found is:
+      # https://github.com/kams-mash/gh-secrets-scanner-action
+      - uses: actions/checkout@v3
+        with:
+          repository: awslabs/git-secrets
+          ref: 1.3.0
+          path: .tmp/git-secrets-repo
+          fetch-depth: 0
+      - name: Compile and install git-secrets
+        run: |
+          cd .tmp/git-secrets-repo
+          sudo make install
+          cd $GITHUB_WORKSPACE
```

Both git and make support `-C somedir` to cd before running commands.  This feels slightly less prone to future error.
```suggestion
          sudo make -C .tmp/git-secrets-repo install
```